content/courses/Ada_For_The_C_Embedded_Developer/chapters/10_Conclusion.rst

```diff
@@ -3,6 +3,175 @@ Conclusion
+faster to write in a condensed manner |mdash| as it is often the case in C.
 
-    Complete chapter!
+Especially in the embedded domain, C developers are used to working at a very
+low level, which includes mathematical operations on pointers, complex bit
+shifts, and logical bitwise operations. C is well designed for such operations
+because it was designed to replace Assembly language for faster, more efficient
+programming.
+
+Ada can be used to describe high level semantics and architectures. The beauty
+of the language, however, is that it can be used all the way down to the lowest
+levels of the development, including embedded Assembly code or bit-level data
+management. However, although Ada supports bitwise operations such as masks and
+shifts, they should be relatively rarely needed. When translating C code to
+Ada, it's good practice to consider alternatives. In a lot of cases, these
+operations are used to insert several pieces of data into a larger structure.
+In Ada, this can be done by describing the structure layout at the type level
+through representation clauses, and then accessing this structure as any other.
+For example, we can interpret an arbitrary data type as a bit-field and perform
+low-level operations on it.
+
+Because Ada is a strongly typed language, it doesn't define any implicit type
+conversions like C. If we try to compile Ada code that contains type
+mismatches, we'll get a compilation error. Because the compiler prevents
+mixing variables of different types without explicit type conversion, we can't
+accidentally end up in a situation where we assume something will happen
+implicitly when, in fact, our assumption is incorrect. In this sense, Ada's
+type system  encourages programmers to think about data at a high level of
+abstraction. Ada supports overlays and unchecked conversions as a way of
+converting between unrelated data type, which are typically used for
+interfacing with low-level elements such as registers.
+
+In Ada, arrays aren't interchangeable with operations on pointers like in C.
+Also, array types are considered first-class citizens and have dedicated
+semantics such as the availability of the array's boundaries at run-time.
+Therefore, unhandled array overflows are impossible unless checks are
+suppressed. Any discrete type can serve as an array index, and we can specify
+both the starting and ending bounds. In addition, Ada offers high-level
+operations for copying, slicing, and assigning values to arrays.
+
+Although Ada supports pointers, most situation that would require a pointer in
+C do not in Ada. In the vast majority of the cases, indirect memory management
+can be hidden from the developer and thus prevent many potential errors. In
+C, pointers are typically used to pass references to subprograms, for example.
+In contrast, Ada supports output parameters (:ada:`out` mode), as well as
+:ada:`in out` parameters.
```

Comment:
I suggest: "In contrast, Ada parameter modes indicate the flow of information to the reader, leaving the means of passing that information to the compiler."  
Otherwise it sounds like the modes are involved in how parameters are passed.